src/Microsoft.AspNetCore.Mvc/DesignTimeMvcServiceCollectionProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    internal class DesignTimeMvcServiceCollectionProvider
```

Comment:
Interesting. Seems inconsistent with our other approaches though. If we took that mindset for the rest of the project we'd have a lot more classes in .Internal